src/Mod/Sketcher/Gui/CommandConstraints.cpp

```diff
@@ -3700,13 +3700,13 @@ void CmdSketcherConstrainPerpendicular::activated(int iMsg)
 
             const Part::Geometry *geom2 = Obj->getGeometry(GeoId2);
 
-            if( geom2 && geom2->getTypeId() == Part::GeomBSplineCurve::getClassTypeId() ){
-                // unsupported until normal to B-spline at any point implemented.
-                Gui::TranslatedNotification(Obj,
-                              QObject::tr("Wrong selection"),
-                              QObject::tr("Perpendicular to B-spline edge currently unsupported."));
-                return;
-            }
+            // if( geom2 && geom2->getTypeId() == Part::GeomBSplineCurve::getClassTypeId() ){
+            //     // unsupported until normal to B-spline at any point implemented.
+            //     Gui::TranslatedNotification(Obj,
+            //                   QObject::tr("Wrong selection"),
+            //                   QObject::tr("Perpendicular to B-spline edge currently unsupported."));
+            //     return;
+            // }
```

Comment:
I've basically removed all restrictions for B-splines right now in GUI. Some may have to be put back up, or need other odifications